core/lib/via_btc_client/src/client/rpc_client.rs

```diff
@@ -1,104 +1,450 @@
-        self.client
-            .get_raw_transaction_info(txid, None)
-            .map_err(|e| e.into())
+        self.with_retry(|| {
+            debug!("Getting raw transaction info");
+            self.client
+                .get_raw_transaction_info(txid, None)
+                .map_err(|e| e.into())
+        })
+        .await
     }
 
+    #[instrument(skip(self), target = "bitcoin_client::rpc_client")]
     async fn estimate_smart_fee(
         &self,
         conf_target: u16,
         estimate_mode: Option<EstimateMode>,
     ) -> BitcoinRpcResult<EstimateSmartFeeResult> {
-        self.client
-            .estimate_smart_fee(conf_target, estimate_mode)
-            .map_err(|e| e.into())
+        self.with_retry(|| {
+            debug!("Estimating smart fee");
+            self.client
+                .estimate_smart_fee(conf_target, estimate_mode)
+                .map_err(|e| e.into())
+        })
+        .await
+    }
+
+    #[instrument(skip(self), target = "bitcoin_client::rpc_client")]
+    async fn get_blockchain_info(&self) -> BitcoinRpcResult<GetBlockchainInfoResult> {
+        self.with_retry(|| {
+            debug!("Getting blockchain info");
+            self.client.get_blockchain_info().map_err(|e| e.into())
+        })
+        .await
     }
 }
 
 #[cfg(test)]
-mod tests {}
+mod tests {
+    use std::str::FromStr;
+
+    use bitcoin::{
+        absolute::LockTime, hashes::sha256d::Hash, transaction::Version, Address, Amount, Block,
+        BlockHash, Network, OutPoint, Transaction, TxMerkleNode, Txid, Wtxid,
+    };
+    use bitcoincore_rpc::json::{EstimateSmartFeeResult, GetRawTransactionResult};
+    use mockall::{mock, predicate::*};
+
+    use super::*;
+
+    mock! {
+        BitcoinRpcClient {}
+
+        #[async_trait]
+        impl BitcoinRpc for BitcoinRpcClient {
+            async fn get_balance(&self, address: &Address) -> BitcoinRpcResult<u64>;
+            async fn send_raw_transaction(&self, tx_hex: &str) -> BitcoinRpcResult<Txid>;
+            async fn list_unspent(&self, address: &Address) -> BitcoinRpcResult<Vec<OutPoint>>;
+            async fn get_transaction(&self, txid: &Txid) -> BitcoinRpcResult<Transaction>;
+            async fn get_block_count(&self) -> BitcoinRpcResult<u64>;
+            async fn get_block_by_height(&self, block_height: u128) -> BitcoinRpcResult<Block>;
+            async fn get_block_by_hash(&self, block_hash: &BlockHash) -> BitcoinRpcResult<Block>;
+            async fn get_best_block_hash(&self) -> BitcoinRpcResult<bitcoin::BlockHash>;
+            async fn get_raw_transaction_info(&self, txid: &Txid) -> BitcoinRpcResult<GetRawTransactionResult>;
+            async fn estimate_smart_fee(&self, conf_target: u16, estimate_mode: Option<EstimateMode>) -> BitcoinRpcResult<EstimateSmartFeeResult>;
+            async fn get_blockchain_info(&self) -> BitcoinRpcResult<GetBlockchainInfoResult>;
+        }
+    }
```

Comment:
What do we test here?
Since you are mocking a struct `BitcoinRpcClient` that should be tested.